.github/workflows/ci.yml

```diff
@@ -50,6 +50,41 @@ jobs:
         with:
           go-version-file: src/go.mod
 
+  # SetM365App will decide which M365 app to use for this CI run
+  SetM365App:
+    environment: Testing
+    runs-on: ubuntu-latest
+    outputs:
+      client_app_slot: ${{ steps.roundrobin.outputs.CLIENT_APP_SLOT }}
+      client_id_env: ${{ steps.roundrobin.outputs.CLIENT_ID_ENV }}
+      client_secret_env: ${{ steps.roundrobin.outputs.CLIENT_SECRET_ENV }}
+    steps:
+      - name: Figure out which client id to use
+        id: roundrobin
+        run: |
+          slot=$((GITHUB_RUN_NUMBER % 4))
```

Comment:
per docs on run_number:
```
A unique number for each run of a particular workflow in a repository. This number begins at 1 for the workflow's first run, and increments with each new run. This number does not change if you re-run the workflow run. For example, 3.
```
If I'm reading that right, it sounds like runs will always kick off on the same number, leading to re-use of the same client id for each new PR.  

We might get something more random with `GITHUB_RUN_ID`
```
A unique number for each workflow run within a repository. This number does not change if you re-run the workflow run. For example, 1658821493.
```

Reply:
I did look at that. My thought was that we actually don't want something random but rather monotonically increasing would give us the behavior we want.